service/dap/server.go

```diff
@@ -415,6 +424,45 @@ func (s *Server) handleRequest(request dap.Message) {
 				Body:     dap.ThreadsResponseBody{Threads: []dap.Thread{{Id: -1, Name: "Current"}}},
 			}
 			s.send(response)
+		case *dap.SetBreakpointsRequest:
+			// Halt running command
+			inProgress := s.debugger.RunningCommand()
+			// The channel is drained before running a command, but just in case drain again to avoid blocking
+			s.resetOverrideStopReason()
+			// TODO(polina): once api.Call is run asynchronously, treat it like api.Continue as well
+			if inProgress == api.Continue {
+				s.overrideStopReason <- skipStop
+			} else {
+				// This would be one of the step commands, which get cancelled by debugger.
```

Comment:
From my reading of the code in proc, when a step /next request is manually halted `ClearInternalBreakpoints` is always called.

It would be great if there is a way to be able to continue next requests, this is also a feature I am interested in for implementing logpoints.

Reply:
We had this conversation a while back when trying to fix setting breakpoints in the old adapter:
https://github.com/golang/vscode-go/issues/787#issuecomment-713434803
I too saw `ClearInternalBreakpoints` and understood it as halt cancelling step.

Reply:
What's a logpoint? Anyway it shouldn't be to hard to make a variant of halt that sets a flag so that ClearInternalBreakpoints isn't called in proc.(*Target).Continue.

Reply:
A log point is like a trace point but with user supplied message

From DAP spec for setBreakpoints arguments:
```
 /**
   * If this attribute exists and is non-empty, the backend must not 'break'
   * (stop)
   * but log the message instead. Expressions within {} are interpolated.
   * The attribute is only honored by a debug adapter if the capability
   * 'supportsLogPoints' is true.
   */
  logMessage?: string;
```